library/Fission/Web/IPFS.hs

```diff
@@ -70,4 +72,16 @@ authed :: HasLogFunc        cfg
 authed usr = CID.allForUser usr
         :<|> Upload.add usr
         :<|> Pin.server usr
-        :<|> DAG.put usr
+        :<|> DAG.put usr -- Question: why is the function naming convention so all over the place?
+                         -- Answer: it's not! This is what they're called in IPFS!
```

Comment:
A fuller answer:

1. This isn't MVC — it's a tree of handlers. Think more Node than Rails.
2. We _could_ structure this as MVC, but it would mainly add boilerplate
  * I'm down to do that if it would help!
3. `server` is like a `controller`, but can be used as a standalone server for testing, handed to clients, &c. The full server is a series of handlers composed with `:<|>` the same way lists are composed with `:` (i.e. `|` in Elixir). Combine two servers with `:<|>`, and you get another server.

That said, if it would help to call it `controller` or `handler`, we can do that, too. Controllers work differently, so that may be additionally confusing.

A lot of these functions are named after what they do, which gives us reuse. The following example adds a layer of direction:

```haskell
module Fission.Storage.IPFS.CID

all = ...

allForUser usr = ...

----------

module Fission.Web.IPFS.CID.Controller where

index = CID.all

---------------

module Fission.Web.IPFS.User.CID.Controller where

index user = CID.allForUser user

------------

module Fission.Web.IPFS where

cidRoutes user = CID.index :<|> User.CID.index user

authed = cidRoutes
    :<|> ...
```

This _does_ give us a familiar interface (CRUD actions), but just adds a bunch of indirection to what could have just been calls to the underlying functions directly.

I'm open to having a conversation about that; it's tradeoffs. In either case we're _just composing functions_ with `:<|>`